File: ios/Detonator/Sources/Detonator/Helper/AttributeHelper.swift
class AttributeHelper {
    static func convertPercentStringToFloat(_ percentString: String) -> Float? {
        let numberString = percentString.replacingOccurrences(of: "%", with: "")
        
        if let value = Float(numberString) {
            return value / 100
        } else {
            return nil
        }
    }
}

File: android/src/main/java/com/iconshot/detonator/helpers/AttributeHelper.java
package com.iconshot.detonator.helpers;

public class AttributeHelper {
    public static Float convertPercentStringToFloat(String percentString) {
        String numberString = percentString.replace("%", "")
                .replace("-", "");

        try {
            float value = Float.parseFloat(numberString);

            float percent = value / 100;

            return percentString.startsWith("-") ? percent * -1 : percent;
        } catch (Exception e) {
            return null;
        }
    }
}
